cpp/src/centrality/betweenness_centrality.cu

```diff
@@ -72,6 +76,21 @@ void edge_betweenness_centrality_impl(raft::handle_t const &handle,
   bc.configure(
     result, is_edge_betweenness, normalize, endpoints, weight, sources, number_of_sources);
   bc.compute();
+  // NOTE: As of 07/2020 NetworkX does not apply rescaling based on number
+  // of sources
+  // bc.rescale_by_total_sources_used(total_number_of_sources);
+}
+template <typename VT>
+VT get_total_number_of_sources(raft::handle_t const &handle, VT local_number_of_sources)
+{
+  VT total_number_of_sources_used = local_number_of_sources;
+  if (handle.comms_initialized()) {
+    rmm::device_vector<VT> d_number_of_sources(1, local_number_of_sources);
```

Comment:
rmm has device_scalar for this purpose (a scalar variable on device).

Look for https://github.com/rapidsai/rmm/blob/branch-0.15/include/rmm/device_scalar.hpp

You need to verify the code below, but I think you can do something like

```
rmm::device_scalar<vertex_t> d_number_of_sources(local_number_of_sources, handle.get_stream());
handle.get_comms().allreduce(d_number_of_souruces.data(), d_number_of_sources.data(), d_number_of_sources.size(), raft::comms::op_t::SUM, handle.get_stream());
totoal_number_of_sources = d_number_of_souruces.value(handle.get_stream());
```

Reply:
Updated in e7c4c4a